theano/configdefaults.py

```diff
@@ -418,6 +419,19 @@ def f(suffix=suffix):
 except OSError:
     rc = 1
 
+# Anaconda on Windows has mingw-w64 packages including GCC, but they may not be on PATH.
+if rc != 0:
+    if "Continuum Analytics" in sys.version and sys.platform == "win32":
+        mingw_w64_gcc = os.path.join(os.path.dirname(sys.executable), "Library", "mingw-w64", "bin", "g++")
+        try:
+            rc = call_subprocess_Popen([mingw_w64_gcc, '-v'])
+            if rc == 0:
+                maybe_add_to_os_environ_pathlist('PATH', os.path.dirname(mingw_w64_gcc))
```

Comment:
I don't understand why you don't force to add the path. If on line 423, rc isn't 0. Then we can't just call g++ without doing something else.

Instead of adding and using the method maybe_add_to_os_environ_pathlist, did you try to replace line 429 to  "param = mingw_w64_gcc" ? param can be a full path to a compiler that support g++ interface. This would be a simpler change that I think would work.